packages/hubble-sdk/src/Hubble.ts

```diff
@@ -511,6 +514,76 @@ export class Hubble {
     globalConfig = replaceBigNumberWithDecimal(globalConfig);
     return globalConfig;
   }
+
+  /**
+   * Get PSM reserve state
+   */
+  async getPsmReserve(): Promise<PsmReserve> {
+    const psmPubkey = await this.getPsmPublicKey();
+    const reserve = (await this._borrowingProgram.account.psmReserve.fetch(psmPubkey)) as PsmReserve;
+    return replaceBigNumberWithDecimal(reserve);
+  }
+
+  /**
+   * Get PSM public key
+   */
+  async getPsmPublicKey(): Promise<PublicKey> {
+    const res = await PublicKey.findProgramAddress(
+      [Buffer.from('PSM'), this._config.borrowing.accounts.USDC!.toBuffer()],
+      this._config.borrowing.programId
+    );
+    return res[0];
+  }
+
+  /**
+   * Get the USDC -> USDH swap information
+   * @param usdcInAmount number of USDC tokens
+   * @param slippage
+   */
+  async getUsdcToUsdhSwap(usdcInAmount: Decimal, slippage: Decimal = new Decimal(0)): Promise<SwapInfo> {
+    const psmReserve = await this.getPsmReserve();
+
+    // we would be minting USDH with this operation
+
+    // remaining USDC = max capacity - deposited,
+    // this is the amount of USDC that can be stored inside the PSM reserve
+    // we can only mint max this much USDH
+    const availableUsdc = psmReserve.maxCapacity
+      .dividedBy(STABLECOIN_DECIMALS)
```

Comment:
Nejc is correct, maxCapacity refers to maxUsdhCapacity afaik, @ignaty8 to confirm

Reply:
so `availableUsdhInLamports = (psmReserve.maxCapacity - psmReserve.mintedUsdh)`
and `availableUsdh = availableUsdhInLamports / usdhDecimals`

Reply:
for USDC, you can use `psmReserve.stablecoinDecimals` instead of a hardcoded const
and you get:
`availableStable= (psmReserve.maxCapacity / usdhDecimals - psmReserve.mintedUsdh / psmReserve.stablecoinDecimals)`

Reply:
if you want it in lamports, you need to calculate a conversion factor USDH -> Stablecoin